Makefile

```
@@ -1,6 +1,6 @@
 SHELL          := bash
 GO             ?= go
-GOOS           ?= $(word 1, $(subst /, " ", $(word 4, $(shell go version))))
+GOOS           ?= $(shell $(GO) version | sed -E 's|^(.* )?([a-zA-Z0-9\-_]+)/[a-zA-Z0-9\-_]+( .*)?$$|\2|')
```

Does this work for you?

```suggestion
GOOS           ?= $(shell $(GO) env GOOS)
```

---

Oh my, I didn't know about this. Yes it seems to work.